### PR TITLE
doc: add apisix_pluginconfig document

### PR DESCRIPTION
**Pull request by neverCase**
### Type of change:

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:
issue: #638 

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by neverCase** [comment]
@tao12345666333

**Review by tao12345666333 — COMMENTED** [review]
thanks

You should modify this file also https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/config.json

**Comment by neverCase** [comment]
> thanks
> 
> You should modify this file also https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/config.json

ok

**Review by tao12345666333 — APPROVED** [review]
LGTM

File: docs/en/latest/references/apisix_pluginconfig.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Spec
+
+Spec describes the desired state of an ApisixPluginConfig object.
+
+|     Field     |  Type    |                    Description                     |
+|---------------|----------|----------------------------------------------------|
+| plugins         | array    | A series of custom plugins that will be executed once this route rule is matched |
+| plugins[].name | string | The plugin name, see [docs](http://apisix.apache.org/docs/apisix/getting-started) for learning the available plugins. |
+| plugins[].enable | boolean | Whether the plugin is in use |
+| plugins[].config | object | The plugin configuration, fields should be same as in APISIX. |
```
Comment (juzhiyuan) [inline docs/en/latest/references/apisix_pluginconfig.md:33]:
```suggestion
| plugins[].config | object | The plugin configuration, and fields should be the same as in APISIX. |
```

File: docs/en/latest/references/apisix_pluginconfig.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Spec
+
+Spec describes the desired state of an ApisixPluginConfig object.
+
+|     Field     |  Type    |                    Description                     |
+|---------------|----------|----------------------------------------------------|
+| plugins         | array    | A series of custom plugins that will be executed once this route rule is matched |
+| plugins[].name | string | The plugin name, see [docs](http://apisix.apache.org/docs/apisix/getting-started) for learning the available plugins. |
+| plugins[].enable | boolean | Whether the plugin is in use |
```
Comment (juzhiyuan) [inline docs/en/latest/references/apisix_pluginconfig.md:32]:
```suggestion
| plugins[].enable | boolean | Whether the plugin should be enabled |
```

Comment (tao12345666333) [reply docs/en/latest/references/apisix_pluginconfig.md:32]: @neverCase PTAL

Comment (juzhiyuan) [reply docs/en/latest/references/apisix_pluginconfig.md:32]: I will merge first, for those small suggestions @neverCase could help to create a new PR :)

Comment (neverCase) [reply docs/en/latest/references/apisix_pluginconfig.md:32]: ok, i will open another one

**Pull request merged**